### PR TITLE
feat: add support for signing keys APIs

### PR DESCRIPTION
**Pull request by tylerburdsall**
Adds support for `CreateSigningKey`, `RevokeSigningKey`, and `ListSigningKeys` APIs

### CODE REVIEW

File: internal/models/responses.go [hunk 1]
```diff
@@ -38,6 +41,54 @@ func NewCacheInfo(cache *pb.XCache) CacheInfo {
...
+type CreateSigningKeyResponse struct {
+	KeyId     string
+	Endpoint  string
+	Key       string
+	ExpiresAt time.Time
+}
+
+func NewCreateSigningKeyResponse(endpoint string, resp *pb.XCreateSigningKeyResponse) *CreateSigningKeyResponse {
+	keyObj := map[string]string{}
+	json.Unmarshal([]byte(resp.GetKey()), &keyObj)
```
Comment (eaddingtonwhite) [inline internal/models/responses.go:57]: for sure want to handle error here and have this function return an error as well.

File: internal/models/responses.go [hunk 2]
```diff
@@ -38,6 +41,54 @@ func NewCacheInfo(cache *pb.XCache) CacheInfo {
...
+
+type CreateSigningKeyResponse struct {
+	KeyId     string
+	Endpoint  string
+	Key       string
+	ExpiresAt time.Time
+}
+
+func NewCreateSigningKeyResponse(endpoint string, resp *pb.XCreateSigningKeyResponse) *CreateSigningKeyResponse {
+	keyObj := map[string]string{}
```
Comment (eaddingtonwhite) [inline internal/models/responses.go:56]: I think you can define this more cleanly like
```go
var keyObj map[string]string
```

File: internal/models/responses.go [hunk 3]
```diff
@@ -38,6 +41,54 @@ func NewCacheInfo(cache *pb.XCache) CacheInfo {
...
+	NextToken string
+}
+
+type ListSigningKeysResponse struct {
+	NextToken   string
+	SigningKeys []SigningKey
+}
+
+func NewListSigningKeysResponse(endpoint string, resp *pb.XListSigningKeysResponse) *ListSigningKeysResponse {
+	var signingKeys = []SigningKey{}
```
Comment (eaddingtonwhite) [inline internal/models/responses.go:75]: same here can just define nil slice like
```go
var signingKeys []SigningKey 
```

should be same

File: internal/models/responses.go [hunk 4]
```diff
@@ -38,6 +41,54 @@ func NewCacheInfo(cache *pb.XCache) CacheInfo {
...
+}
+
+type SigningKey struct {
+	KeyId     string
+	Endpoint  string
+	ExpiresAt time.Time
+}
+
+func NewSigningKey(endpoint string, signingKey *pb.XSigningKey) SigningKey {
+	return SigningKey{KeyId: signingKey.GetKeyId(), Endpoint: endpoint, ExpiresAt: time.Unix(int64(signingKey.GetExpiresAt()), 0)}
```
Comment (eaddingtonwhite) [inline internal/models/responses.go:89]: This just me personally but I like expanding stuff like this out I think easier to read. Hard to quickly see fields on object being returned when all on one line. Take as you will dont feel super strongly on this.

```go
return SigningKey{
  KeyId: signingKey.GetKeyId(), 
  Endpoint: endpoint, 
  ExpiresAt: time.Unix(int64(signingKey.GetExpiresAt()), 0)
}
```

File: internal/utility/utility.go [hunk 5]
```diff
@@ -23,3 +23,10 @@ func IsValueValid(value interface{}) momentoerrors.MomentoSvcErr {
 	}
 	return nil
 }
+
+func IsTtlValid(ttl uint64) momentoerrors.MomentoSvcErr {
+	if ttl < 0 {
```
Comment (eaddingtonwhite) [inline internal/utility/utility.go:28]: like lint error said unsigned int cannot be less then 0 I did think we wanted to try and fail on 0 ttl moving forward though in sdks so maybe can make `ttl < 1`. Erica can confitm.
cc/ @poppoerika

File: momento/simple_cache_client.go [hunk 6]
```diff
@@ -150,6 +154,57 @@ func (c *DefaultScsClient) ListCaches(request *ListCachesRequest) (*ListCachesRe
 	}, nil
 }
 
+// Creates a Momento signing key in your Moemnto account
```
Comment (eaddingtonwhite) [inline momento/simple_cache_client.go:157]: Moemnto

File: momento/simple_cache_client.go [hunk 7]
```diff
@@ -150,6 +154,57 @@ func (c *DefaultScsClient) ListCaches(request *ListCachesRequest) (*ListCachesRe
...
+	}
+	return &CreateSigningKeyResponse{
+		keyId:     rsp.KeyId,
+		endpoint:  rsp.Endpoint,
+		key:       rsp.Key,
+		expiresAt: rsp.ExpiresAt,
+	}, nil
+}
+
+// Revokes a Momento signing key in your Moemnto account, all tokens signed by which will be invalid
```
Comment (eaddingtonwhite) [inline momento/simple_cache_client.go:177]: Moemnto

File: momento/simple_cache_client.go [hunk 8]
```diff
@@ -150,6 +154,57 @@ func (c *DefaultScsClient) ListCaches(request *ListCachesRequest) (*ListCachesRe
...
+	err := c.controlClient.RevokeSigningKey(&models.RevokeSigningKeyRequest{
+		KeyId: request.KeyId,
+	})
+	if err != nil {
+		return convertMomentoSvcErrorToCustomerError(err)
+	}
+	return nil
+}
+
+// Lists all Momento signing keys in your Moemnto account
```
Comment (eaddingtonwhite) [inline momento/simple_cache_client.go:191]: Moemnto

**Review by eaddingtonwhite — DISMISSED** [review]
@tylerburdsall this is awesome!! ❤️ were adding this to go sdk now also. Left some feed back but overall looking good. 

Could we also try running golint and gofmt locally and check out put. It seems like were not totally compliant to me but hard to say for sure. Not sure if we need an extra cicd step maybe need to audit.

File: internal/utility/utility.go (hunk 5 above)
Comment (tylerburdsall) [reply internal/utility/utility.go]: 🤦 thank you!

**Review by wandaitzuchen — APPROVED** [review]

**Pull request merged**